coil-gif/src/main/java/coil3/gif/AnimatedImageDecoder.kt

```diff
@@ -54,10 +56,22 @@ class AnimatedImageDecoder(
 
     override suspend fun decode(): DecodeResult {
         var isSampled = false
+
+        val loopCount = if (options.repeatCount == GIF_LOOP_COUNT) {
+            extractLoopCountFromGif(source.source().peek()) ?: REPEAT_INFINITE
```

Comment:
@revonateB0T 

Ideally, it would be best to parse the Loop Count directly during the initial parsing of the GIF. However, currently, I have concluded that we need to use either `peek()` or `clone()` of `BufferedSource`.

Between these two methods, would it be better to use `clone()` to allow for garbage collection of the memory, rather than using `peek()`? 

Or is the approach itself flawed?

I am quite concerned that the current implementation may not be the most efficient in terms of performance and memory management. I would appreciate any appropriate advice you could provide.

Reply:
ㅅ

> You cannot clone a `BufferedSource` to another `BufferedSource`, as `Source` is one-shot. It's hard to optimize it currently if we consist on that `BufferedSource` abstraction. We need to break that, consider a ByteBuffer source which is seekable & **rewindable**, a File is seekable & **rewindable**, so I suggest you warn user we may OOM, let they know the trade-off.

Thank you so much for your advice. Let's think more about this !!